src/cellars/cellar_v2.rs

```diff
@@ -3,34 +3,32 @@
 //! To learn more see https://github.com/PeggyJV/cellar-contracts/blob/main/src/base/Cellar.sol
 use abscissa_core::tracing::{debug, info};
 use ethers::{abi::AbiEncode, contract::EthCall, types::Bytes};
-use GovernanceFunction::*;
-use StrategyFunction::*;
 
 use crate::{
     abi::cellar_v2::{AdaptorCall as AbiAdaptorCall, *},
     cellars::adaptors,
     error::Error,
     proto::{
-        adaptor_call::CallData::*, cellar_v2::Function as StrategyFunction,
+        adaptor_call::CallData::*, cellar_v2::Function,
         cellar_v2_governance::Function as GovernanceFunction, AdaptorCall,
     },
     utils::{sp_call_error, sp_call_parse_address, string_to_u256},
 };
 
 use super::{
-    check_blocked_adaptor, check_blocked_position, log_cellar_call, log_governance_cellar_call,
+    check_blocked_adaptor, log_cellar_call, log_governance_cellar_call, validate_new_adaptor,
+    V2_0_PERMISSIONS,
 };
 
 const CELLAR_NAME: &str = "CellarV2";
 
-pub fn get_encoded_call(function: StrategyFunction, cellar_id: String) -> Result<Vec<u8>, Error> {
+pub fn get_encoded_call(function: Function, cellar_id: String) -> Result<Vec<u8>, Error> {
     get_call(function, cellar_id).map(|call| call.encode())
 }
 
-pub fn get_call(function: StrategyFunction, cellar_id: String) -> Result<CellarV2Calls, Error> {
+pub fn get_call(function: Function, cellar_id: String) -> Result<CellarV2Calls, Error> {
     match function {
-        AddPosition(params) => {
-            check_blocked_position(&params.position_id)?;
+        Function::AddPosition(params) => {
```

Comment:
We need to add the checks we've ported here:

`validate_new_position(&cellar_id, params.position_id, &V2_0_PERMISSIONS)?;`